### PR TITLE
Add support for SEVIRI data from icare

### PR DESCRIPTION
**Pull request by simonrp84**
The [ICARE](http://www.icare.univ-lille1.fr/) service has the full record of data for many satellites at level 1, including SEVIRI.
This PR adds support for ICARE / SEVIRI data in HDF4 format, which can be accessed with `Scene(files, reader='seviri_l1b_icare')`

There is one HDF file per band and data is padded, meaning that `HRV` data is projected onto a `11136x11136` array rather than being subset into North and South sectors.

 - [x] Tests added and test suite added to parent suite 
 - [x] Tests passed
 - [x] Passes ``flake8 satpy`` 
 - [x] Fully documented 
`

### CODE REVIEW

File: satpy/readers/seviri_l1b_icare.py [hunk 1]
```diff
@@ -0,0 +1,276 @@
...
+                    'sensor': self.sensor_name[1],
+                    'zone': self.zone,
+                    'satellite_altitude': self.alt,
+                    'cfac': geoloc[0],
+                    'lfac': geoloc[1],
+                    'coff': geoloc[2],
+                    'loff': geoloc[3],
+                    'resolution': self.res,
+                    'satellite_longitude': self.satlon,
+                    'projection_longitude': self.projlon,
```
Comment (mraspaud) [inline satpy/readers/seviri_l1b_icare.py]: We have standardised the way to provide the different satellite longitudes here : https://satpy.readthedocs.io/en/latest/readers.html#metadata

**Comment by simonrp84** [comment]
> I know this is still a draft, but here are some preliminary comments :)

Thanks Martin - I wasn't aware of the longitude standards. All correct now :)

File: satpy/readers/seviri_l1b_icare.py [hunk 2]
```diff
@@ -0,0 +1,273 @@
...
+# Foundation, either version 3 of the License, or (at your option) any later
+# version.
+#
+# satpy is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
+# A PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along with
+# satpy.  If not, see <http://www.gnu.org/licenses/>.
+r"""Interface to SEVIRI L1B data from ICARE (Lille).
```
Comment (mraspaud) [inline satpy/readers/seviri_l1b_icare.py]: Could you add a link to the format description (at the bottom of the docstring?) ?

Comment (simonrp84) [reply satpy/readers/seviri_l1b_icare.py]: Unfortunately no documentation is available, but if it becomes available in the future I will add it.

**Comment by simonrp84** [comment]
I talked to icare / Lille about documentation and their reply is that because HDF is a self-documenting file format there is no need for external documentation, so that means I can't add a link to any documentation!
Aside from that, I think this PR is ready now.

**Pull request merged**